### PR TITLE
feat: detect renamed and duplicate images

### LINKED ISSUE #10: feat: do not re-index an image if it was renamed
Compute and store image hashes, and add logic to handle image renaming properly.

### PR DESCRIPTION
**Pull request by yurijmikhalevich**
## How does this PR impact the user?

Detects renamed and duplicate images, reusing vectors instead of re-indexing. Users get faster consecutive runs when files are renamed or duplicated.

## Description

- [x] add SHA-256 hash column to images table (DB version 3 → 4)
- [x] compute hash during file read (first 1MB + last 1KB, ~0.5ms overhead)
- [x] check for hash matches before CLIP inference; reuse vector if match found
- [x] add tests for rename and duplicate detection

## Limitations

- requires initial indexing to populate hashes
- hash is O(1) but adds ~0.5ms per new image (<2% overhead)

## Checklist

- [x] my PR is focused and contains one holistic change
- [ ] I have added screenshots or screen recordings to show the changes

Closes #10

### CODE REVIEW

File: rclip/db.py [hunk 1]
```diff
@@ -106,10 +112,10 @@ def _get_dirpath_like_pattern(path: str) -> str:
...
-      INSERT INTO images(deleted, indexing, filepath, modified_at, size, vector)
-      VALUES (:deleted, :indexing, :filepath, :modified_at, :size, :vector)
+      INSERT INTO images(deleted, indexing, filepath, modified_at, size, vector, hash)
+      VALUES (:deleted, :indexing, :filepath, :modified_at, :size, :vector, :hash)
       ON CONFLICT(filepath) DO UPDATE SET
-        deleted=:deleted, indexing=:indexing, modified_at=:modified_at, size=:size, vector=:vector
+        deleted=:deleted, indexing=:indexing, modified_at=:modified_at, size=:size, vector=:vector, hash=:hash
     """,
       {"deleted": None, "indexing": None, **image},
     )
```
[the remark this answers is not in the document]
Comment (yurijmikhalevich) [reply rclip/db.py:121]: All callers are already passing the `hash`.

File: tests/e2e/test_rclip.py [hunk 2]
```diff
@@ -378,3 +378,67 @@ def test_unicode_filepaths(
...
+  from rclip.main import init_rclip
+
+  with tempfile.TemporaryDirectory() as temp_dir:
+    temp_path = Path(temp_dir)
+
+    cat_image = temp_path / "cat.jpg"
+    shutil.copy2(test_images_dir / "cat.jpg", cat_image)
+
+    rclip, _, db = init_rclip(str(temp_path), 8)
+
```
[the remark this answers is not in the document]
Comment (yurijmikhalevich) [reply tests/e2e/test_rclip.py]: https://github.com/yurijmikhalevich/rclip/pull/299/commits/cfc50eb96ea30570312140a771868fca196f8942

File: rclip/utils/helpers.py [hunk 3]
```diff
@@ -370,6 +371,22 @@ def read_raw_image_file(path: str):
...
+  Combined with file size check (already done), collision probability is essentially zero.
+  """
+  hasher = hashlib.sha256()
+  with open(path, "rb") as f:
+    hasher.update(f.read(1024 * 1024))
+    file_size = f.seek(0, 2)
+    if file_size > 1024:
+      f.seek(-1024, 2)
+      hasher.update(f.read(1024))
+  return hasher.hexdigest()
```
[the remark this answers is not in the document]
Comment (yurijmikhalevich) [reply rclip/utils/helpers.py]: https://github.com/yurijmikhalevich/rclip/pull/299/commits/cfc50eb96ea30570312140a771868fca196f8942

File: rclip/db.py [hunk 4]
```diff
@@ -146,6 +152,13 @@ def get_image(self, **kwargs: Any) -> Optional[Image]:
     cur = self._con.execute(f"SELECT * FROM images WHERE {query} LIMIT 1", kwargs)
     return cur.fetchone()
 
+  def get_images_by_hash(self, hash_value: str) -> list[Image]:
+    cur = self._con.execute(
+      "SELECT * FROM images WHERE hash = ? AND deleted IS NULL",
+      (hash_value,),
+    )
+    return cast(list[Image], [dict(row) for row in cur.fetchall()])
```
[the remark this answers is not in the document]
Comment (yurijmikhalevich) [reply rclip/db.py:160]: Looks like it works.
